packages/react-router-dom/dom.ts

```diff
@@ -97,13 +97,18 @@ export function getSearchParamsForLocation(
   let searchParams = createSearchParams(locationSearch);
 
   if (defaultSearchParams) {
-    for (let key of defaultSearchParams.keys()) {
+    // Use `defaultSearchParams.forEach(...)` here instead of iterating of
+    // `defaultSearchParams.keys()` to work-around a bug in Firefox related to
+    // web extensions. Relevant Bugzilla tickets:
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1414602
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1023984
+    defaultSearchParams.forEach((_, key) => {
       if (!searchParams.has(key)) {
         defaultSearchParams.getAll(key).forEach((value) => {
           searchParams.append(key, value);
         });
       }
-    }
+    });
   }
```

Comment:
Since we're at it, let's introduce optional chaining in this piece of code as well

```suggestion
  // Use `defaultSearchParams.forEach(...)` here instead of iterating of
  // `defaultSearchParams.keys()` to work-around a bug in Firefox related to
  // web extensions. Relevant Bugzilla tickets:
  // https://bugzilla.mozilla.org/show_bug.cgi?id=1414602
  // https://bugzilla.mozilla.org/show_bug.cgi?id=1023984
  defaultSearchParams?.forEach((_, key) => {
    if (!searchParams.has(key)) {
      defaultSearchParams.getAll(key).forEach((value) => {
        searchParams.append(key, value);
      });
    }
  });
```

Reply:
We're trying to avoid optional chaining in RR code due to bundle bloat concerns :/